packages/ui/src/components/vuestic-components/va-pagination/VaPagination.vue

```diff
@@ -1,215 +1,258 @@
 <template>
-  <va-button-group class="va-pagination">
+  <va-button-group
+    class="va-pagination"
+    v-if="showPagination"
+  >
     <va-button
-      v-if="this.pages !== visiblePages && boundaryLinks"
-      :style="directionButtonStyle"
+      v-if="showBoundaryLinks"
       outline
       :color="color"
       :size="size"
       :disabled="disabled || value === 1"
       :icon="iconClass.boundary"
-      icon-right=""
       @click="changePage(1)"
+      :flat="flat"
     />
     <va-button
-      v-if="this.pages !== visiblePages && directionLinks"
-      :style="directionButtonStyle"
+      v-if="showDirectionLinks"
       outline
       :color="color"
       :size="size"
       :disabled="disabled || value === 1"
       :icon="iconClass.direction"
-      icon-right=""
       @click="changePage(value - 1)"
+      :flat="flat"
     />
-    <va-button
-      :style="activeButtonStyle(n)"
-      outline
+    <slot v-if="!input">
+      <va-button
+        :style="activeButtonStyle(n)"
+        outline
+        v-for="(n, key) in paginationRange"
+        :key="key"
+        :color="color"
+        :size="size"
+        :disabled="disabled || n === '...'"
+        :class="{
+          'va-button--active': n === value,
```

Comment:
Here's how I would name it:

`va-pagination__va-button--active`
Normally we use component name as bem block only in component context.